src/main/java/org/apache/maven/buildcache/RemoteCacheRepositoryImpl.java

```diff
@@ -147,50 +157,33 @@ public void saveArtifactFile( CacheResult cacheResult,
     @Nonnull
     public Optional<byte[]> getResourceContent( String url ) throws IOException
     {
-        HttpGet get = new HttpGet( url );
         try
         {
             LOGGER.info( "Downloading {}", url );
-            HttpResponse response = httpClient.get().execute( get );
-            int statusCode = response.getStatusLine().getStatusCode();
-            if ( statusCode != HttpStatus.SC_OK )
-            {
-                LOGGER.info( "Cannot download {}, status code: {}", url, statusCode );
-                return Optional.empty();
-            }
-            try ( InputStream content = response.getEntity().getContent() )
-            {
-                return Optional.of( IOUtils.toByteArray( content ) );
-            }
+            GetTask task = new GetTask( new URI( url ) );
+            transporter.get( task );
+            return Optional.of( task.getDataBytes() );
```

Comment:
Word of warning: unsure the data size here, but if not "heap friendly" sized, this is OOM prone